File: scripts/algorithms/S/Sum of Even Numbers After Queries/Sum of Even Numbers After Queries.py
# Runtime: 1565 ms (Top 5.26%) | Memory: 20.6 MB (Top 46.36%)
class Solution:
# O(n) || O(1)
# Runtime: 583ms 72.40% || memory: 20.5mb 37.69%
    def sumEvenAfterQueries(self, nums: List[int], queries: List[List[int]]) -> List[int]:
        totalEvenNumSum = sum([num for num in nums if num % 2 == 0])
        result = []

        for val, idx in queries:
            oldVal = nums[idx]
            nums[idx] += val

            if oldVal % 2 == 0:
                totalEvenNumSum -= oldVal

            if nums[idx] % 2 == 0:
                totalEvenNumSum += nums[idx]

            result.append(totalEvenNumSum)

        return result

File: scripts/algorithms/S/Sum of Even Numbers After Queries/Sum of Even Numbers After Queries.js
// Runtime: 187 ms (Top 62.12%) | Memory: 50.6 MB (Top 51.52%)
/**
 * @param {number[]} nums
 * @param {number[][]} queries
 * @return {number[]}
 */
var sumEvenAfterQueries = function(nums, queries) {
    let sumEven = 0, ans = []
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] % 2 == 0) sumEven += nums[i]
    }

    for (let [val, idx] of queries) {
        if (nums[idx] % 2 == 0) {
            if (val % 2 == 0) sumEven += val
            else sumEven -= nums[idx]
        } else {
            // odd + odd
            if (val % 2 !== 0) sumEven += nums[idx] + val
        }
        nums[idx] += val
        ans.push(sumEven)
    }

    return ans
};